cranelift/codegen/src/machinst/lower.rs

```diff
@@ -172,17 +163,22 @@ pub trait LowerCtx {
     fn ensure_in_vreg(&mut self, reg: Reg, ty: Type) -> Reg;
 }
 
-/// A representation of all of the ways in which an instruction input is
-/// available: as a producing instruction (in the same color-partition), as a
-/// constant, and/or in an existing register. See [LowerCtx::get_input] for more
-/// details.
+/// A representation of all of the ways in which a value is available, aside
+/// from as a direct register.
+///
+/// - An instruction, if it would be allowed to occur at the current location
+///   instead (see [LowerCtx::maybe_get_input_as_source_or_const()] for more
+///   details).
+///
+/// - A constant, if the value is known to be a constant.
 #[derive(Clone, Copy, Debug)]
 pub struct LowerInput {
```

Comment:
I feel like this could use a new name now: `NonRegInput`?

Reply:
I like that name better, agreed.